src/cript/nodes/core.py

```diff
@@ -150,6 +150,8 @@ def validate(self, api=None) -> None:
 
     @classmethod
     def _from_json(cls, json_dict: dict):
+        # TODO find a way to handle uuid nodes only
```

Comment:
this is me being picky, I would say if you can make this more descriptive so anyone looking at it knows what you are talking about and can know to work around it or possibly help with it. 

If it would be too long to write out then it is no big deal either, it is just a comment

Reply:
yeah, maybe. I am not sure if I want it even there.

The new JSON hook class might be better for this.

Reply:
hmm, that is a good point.

Before upgrading the serializer and reserializer lets be sure to walk through it a bit because if we end up switching to Pydantic it could end up doing a lot of the heavy lifting for us